### ISSUE TITLE
build images only for master branch

### ISSUE THREAD

**Issue by KarolAdamiec** [Shippable/support#2849]
I want my CI to test PRs. But build only images of master. And only after they are actually merged in.
How do i set it up?
My current yaml:
```yaml
language: node_js

node_js:
    - 5.11

build:
    ci:
    - echo "is a PR? $IS_PULL_REQUEST"
    - cd client; npm install
    - npm test
    - npm run build
    - cd ../server; npm install
    - echo "module.exports =  $BUILD_NUMBER;" > dist/version.js
    - npm test
    - cd ..
    post_ci:
    -  if [ "$IS_PULL_REQUEST" == false ]; then docker build -t trilogiq/web-template .; fi
    -  if [ "$IS_PULL_REQUEST" == false ]; then docker tag -f trilogiq/web-template:latest 975150373617.dkr.ecr.eu-west-1.amazonaws.com/trilogiq/web-template:$BUILD_NUMBER; fi
    -  if [ "$IS_PULL_REQUEST" == false ]; then docker push 975150373617.dkr.ecr.eu-west-1.amazonaws.com/trilogiq/web-template:$BUILD_NUMBER; fi
integrations:
  hub:
    - integrationName: ecr_repository
      type: ecr
      region: eu-west-1
      branches:
        only:
          - master
```

Problem is that after i merge PR into master another build is not kicked off. So the end result is no image.

**Comment by KarolAdamiec** [comment]
AS a side note, enabling web hook triggers to build also commits and not only PRs makes the final PR merge into master build which is what i want. I am not interested though in having CI run on every commits. As it stands i either choose PR or commit trigers and fumble with variables??

**Comment by a-murphy** [comment]
If your pull requests are only to master, you could add a [branches: only](http://docs.shippable.com/ci_configure/#includingexcluding-branches) section to your shippable.yml specifying that only master should be built.  Then with both commit and pull request builds enabled, only the pull requests to master and commits on master would be built.

**Comment by KarolAdamiec** [comment]
Thank you. That is exactly the setup that i ended up with just hour ago.
This can be closed.

One minor thing is that CI build documentation section is absolutely not easy to spot that branch:only can be applied also at root level of shippable.yml.
